File: adjacentElementsProduct.ts
function adjacentElementsProduct(numbers: number[]): number {
  let largestProduct = numbers[0] * numbers[1];
  for (let i = 1; i < numbers.length - 1; i++) {
    const product = numbers[i] * numbers[i + 1];
    largestProduct = largestProduct < product ? product : largestProduct;
  }
  return largestProduct;
}

console.log(adjacentElementsProduct([10, 1, 13, 2, 3, 25, 1, 100]));

File: AdjacentElementsProduct.java
public class AdjacentElementsProduct {

    public static void  main(String[] args){
        System.out.println(AdjacentElementsProduct(new int[]{3, 6, -2, -5, 7, 3,30}));
    }
    public  static  int AdjacentElementsProduct(int [] arr){

        int product = 0;

        for(int i= 0; i < arr.length-1;i++){
            if (arr[i] * arr[i+1] > product){
                product = arr[i] * arr[i+1];
            }
        }
        return product;
    }
}
